packages/graph/hash_graph/lib/graph/src/ontology/property_type.rs

```diff
@@ -26,7 +31,243 @@ impl QueryRecord for PropertyType {
+                    _ => Err(de::Error::invalid_value(Unexpected::Str(v), &self)),
+                }
+            }
+
+            fn visit_bytes<E: de::Error>(self, v: &[u8]) -> Result<Self::Value, E> {
+                match core::str::from_utf8(v) {
+                    Ok(s) => self.visit_str(s),
+                    Err(_) => Err(E::invalid_value(de::Unexpected::Bytes(v), &self)),
+                }
+            }
+        }
+
+        deserializer.deserialize_str(SelectorVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::ontology::test_utils::create_path;
+
+    fn convert_path(
+        segments: impl IntoIterator<Item = &'static str>,
+    ) -> PropertyTypeQueryPath<'static> {
+        PropertyTypeQueryPath::try_from(create_path(segments)).expect("Could not convert path")
+    }
+
+    fn deserialize<'q>(segments: impl IntoIterator<Item = &'q str>) -> PropertyTypeQueryPath<'q> {
+        PropertyTypeQueryPath::deserialize(de::value::SeqDeserializer::<_, de::value::Error>::new(
+            segments.into_iter(),
+        ))
+        .expect("Could not deserialize path")
+    }
+
+    #[test]
+    fn deserialization() {
+        assert_eq!(deserialize(["baseUri"]), PropertyTypeQueryPath::BaseUri);
+        assert_eq!(deserialize(["version"]), PropertyTypeQueryPath::Version);
+        assert_eq!(
+            deserialize(["versionedUri"]),
+            PropertyTypeQueryPath::VersionedUri
+        );
+        assert_eq!(deserialize(["ownedById"]), PropertyTypeQueryPath::OwnedById);
+        assert_eq!(deserialize(["title"]), PropertyTypeQueryPath::Title);
+        assert_eq!(
+            deserialize(["description"]),
+            PropertyTypeQueryPath::Description
+        );
+        assert_eq!(
+            deserialize(["dataTypes", "*", "version"]),
+            PropertyTypeQueryPath::DataTypes(DataTypeQueryPath::Version)
+        );
+        assert_eq!(
+            deserialize(["propertyTypes", "*", "baseUri"]),
+            PropertyTypeQueryPath::PropertyTypes(Box::new(PropertyTypeQueryPath::BaseUri))
+        );
+
+        assert_eq!(
+            PropertyTypeQueryPath::deserialize(
+                de::value::SeqDeserializer::<_, de::value::Error>::new(
+                    ["baseUri", "test"].into_iter()
+                )
+            )
+            .expect_err("Could convert property type query path with multiple tokens")
+            .to_string(),
+            "invalid length 2, expected 1 element in sequence"
+        );
+
+        assert_eq!(
+            PropertyTypeQueryPath::deserialize(
+                de::value::SeqDeserializer::<_, de::value::Error>::new(
+                    ["dataTypes", "*"].into_iter()
+                )
+            )
+            .expect_err("Could convert property type query path with multiple tokens")
+            .to_string(),
+            "invalid length 2, expected one of `ownedById`, `baseUri`, `versionedUri`, `version`, \
```

Comment:
These are the sorts of things that seem really annoying to keep track of and update with any changes to the message format.

Maybe we just want const strings for the error messages so we can use format strings?

Reply:
I made it a constant in 41f4838615d7d7ce32e1e0c0470eef4c016cca85